api/chaos/experimentation/v1/experimentation.proto

```diff
@@ -2,21 +2,34 @@ syntax = "proto3";
 
 package clutch.chaos.experimentation.v1;
 
+import "api/v1/annotations.proto";
 import "google/api/annotations.proto";
 import "google/protobuf/any.proto";
+import "google/protobuf/timestamp.proto";
 import "validate/validate.proto";
 
-import "api/v1/annotations.proto";
-
 option go_package = "experimentationv1";
 
+message ExperimentParams {
+
+  // The experiment configuration specific to the type of experiment
+  google.protobuf.Any config = 2;
+
+  // The time when the experiment should start. If unspecified, defaults to 'now'
+  google.protobuf.Timestamp start_time = 3;
+
+  // The time when the experiment should end, If unspecified, the experiment runs indefinitely.
+  google.protobuf.Timestamp end_time = 4;
+}
+
 message Experiment {
   fixed64 id = 1;
-  google.protobuf.Any config = 2;
+
+  ExperimentParams params = 2;
```

Comment:
If we change `Experiment` type we will also change the look of the Experiment that are returned from this method (which will break existing usages of `GetExperiments`):

For example:
before change:
```
Experiment:
    id: 123
    config: {}
```

After change
```
Experiment:
   id: 123
   experiment_params: 
       config: {}
``

From the perspective of the user of this API, the second proto looks harder to use - it uses nested `experiment_params` type that users of this api should not really need to care about.

Reply:
I think that your initial idea to just introduce a separate proto for experiment creation would result in a "nicer to use API" here:

```proto
message ExperimentCreationParams {
    int64 id = 1
    google.protobuf.Timestamp start_time = 2;
    google.protobuf.Timestamp end_time = 3;
    // ...
}

message CreateExperimentsRequest {
   repeated ExperimentCreationParams experiments = 1
}

message CreateExperimentsResponse {
  repeated Experiment experiments = 1;
}
```

With this API, a user quering the list of experiments receives the following "JSON" (notice the flat structure of the returned object, `config` is not nested):
```
Experiment:
   id: 123
   config: {}
```

Reply:
but that gets us back to the problem that @danielhochman was trying to avoid: passing an uninitialized id to the create function.

Reply:
this is the comment I'm referring to: https://github.com/lyft/clutch/pull/336#discussion_r478695477

Reply:
The more I think about the more I like the proposed approach ^. Lets avoid coupling of request and response data. Pretty soon, our `GetExperimentsResponse` will need to start returning fields such as `termination_reason` or `scheduled_end_time` - both of them should *not* be a part of `CreateExperimentRequest`

Reply:
another option is to only support creating a single experiment at a time. then we don't need a struct to carry the params, they can be defined in CreateExperimentRequest.

Reply:
@wgallagher my bad - `ExperimentCreationParams` doesn't have to have `id` field.